hummingbot/market/hitbtc/hitbtc_active_order_tracker.pyx

```diff
@@ -0,0 +1,163 @@
+# distutils: language=c++
+# distutils: sources=hummingbot/core/cpp/OrderBookEntry.cpp
+
+import logging
+
+import numpy as np
+from decimal import Decimal
+from typing import Dict
+
+from hummingbot.logger import HummingbotLogger
+from hummingbot.core.data_type.order_book_row import OrderBookRow
+
+_hbaot_logger = None
+s_empty_diff = np.ndarray(shape=(0, 4), dtype="float64")
+
+HitbtcOrderBookTrackingDictionary = Dict[Decimal, Dict[str, Dict[str, any]]]
+
+cdef class HitbtcActiveOrderTracker:
+    def __init__(self,
+                 active_asks: HitbtcOrderBookTrackingDictionary = None,
+                 active_bids: HitbtcOrderBookTrackingDictionary = None):
+        super().__init__()
+        self._active_asks = active_asks or {}
+        self._active_bids = active_bids or {}
+
+    @classmethod
+    def logger(cls) -> HummingbotLogger:
+        global _hbaot_logger
+        if _hbaot_logger is None:
+            _hbaot_logger = logging.getLogger(__name__)
+        return _hbaot_logger
+
+    @property
+    def active_asks(self) -> HitbtcOrderBookTrackingDictionary:
+        return self._active_asks
+
+    @property
+    def active_bids(self) -> HitbtcOrderBookTrackingDictionary:
+        return self._active_bids
+
+    def volume_for_ask_price(self, price) -> float:
+        return sum([float(msg["remaining_size"]) for msg in self._active_asks[price].values()])
+
+    def volume_for_bid_price(self, price) -> float:
+        return sum([float(msg["remaining_size"]) for msg in self._active_bids[price].values()])
+
+    def get_rates_and_quantities(self, entry) -> tuple:
+        return entry["R"], entry["Q"]
```

Comment:
Refer to this [comment](https://github.com/CoinAlpha/hummingbot/pull/1039/files#r343463345).